pkg/minikube/bootstrapper/kubeadm/kubeadm.go

```diff
@@ -335,7 +335,8 @@ func (k *Bootstrapper) applyCNI(cfg config.ClusterConfig) error {
 		return errors.Wrap(err, "cni apply")
 	}
 
-	if cfg.KubernetesConfig.ContainerRuntime == constants.CRIO {
+	_, custom := cnm.(cni.Custom)
+	if cfg.KubernetesConfig.ContainerRuntime == constants.CRIO && !custom {
```

Comment:
I am curious why only crio, how about containerd, wouldn't we need to remove the the kindnet cni if user specifies the custom cni ?

Reply:
If user uses a custom cni, the `kindnet.conflist` file shouldn't be created. But even if kindnet is used the delay in config file creation is there. For CRIO - the kind kindnet file is created almost 17 seconds after the crio-bridge and podman config file. See - 

```
root@minikube:/etc/cni/net.d# stat 10-kindnet.conflist
  File: 10-kindnet.conflist
  Size: 391       	Blocks: 8          IO Block: 4096   regular file
Device: 74h/116d	Inode: 3288136     Links: 1
Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2021-02-26 22:36:37.270550000 +0000
Modify: 2021-02-26 22:36:37.241550000 +0000
Change: 2021-02-26 22:36:37.243550000 +0000
 Birth: -


root@minikube:/etc/cni/net.d# stat 100-crio-bridge.conf
  File: 100-crio-bridge.conf
  Size: 438       	Blocks: 8          IO Block: 4096   regular file
Device: 74h/116d	Inode: 3288094     Links: 1
Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2021-02-26 22:36:20.385550000 +0000
Modify: 2021-02-26 22:36:20.356550000 +0000
Change: 2021-02-26 22:36:20.356550000 +0000
 Birth: -


root@minikube:/etc/cni/net.d# stat 87-podman-bridge.conflist
  File: 87-podman-bridge.conflist
  Size: 641       	Blocks: 8          IO Block: 4096   regular file
Device: 74h/116d	Inode: 3288095     Links: 1
Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2021-02-26 22:36:20.365550000 +0000
Modify: 2021-02-26 22:36:20.356550000 +0000
Change: 2021-02-26 22:36:20.356550000 +0000
 Birth: -
```

But probably we do not need to delete other config files - instead a kubelet should start after the file is created. https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/#cni.

Reply:
The podman and cri-o CNI config is installed with the package, so they are there at boot.

The podman configuration is also totally unrelated to Kubernetes, even though it uses CNI

Reply:
I don't think that it is unrelated. When --cni is specified /etc/cni/net.d/1-k8s.conf is NOT created which only leaves 87-podman-bridge.conflist in /etc/cni/net.d. In absence of any other cni config, the kubelet uses the podman cni to assign IPs to Pods which come up BEFORE the config of the 3rd party is copied. 
Hence the coredns is served by podman cni. Manually deleting the coredns Pod solves the problem but that's not really ideal.
Removing/not creating  87-podman-bridge.conflist solves that problem: coredns Pod is only created when the kubelet finds the 3rd party cni config and everything is fine.
And yes, podman run now has a problem.